### ISSUE TITLE
SqueezeNAS: Fast neural architecture search for faster semantic segmentation

### ISSUE THREAD

**Issue by sile** [sile/readings#153]
URL
-----

https://arxiv.org/abs/1908.01748

Abstract
----------

> For real time applications utilizing Deep Neural Networks (DNNs), it is critical that the models achieve high-accuracy on the target task and low-latency inference on the target computing platform. While Neural Architecture Search (NAS) has been effectively used to develop low-latency networks for image classification, there has been relatively little effort to use NAS to optimize DNN architectures for other vision tasks. In this work, we present what we believe to be the first proxyless hardware-aware search targeted for dense semantic segmentation. With this approach, we advance the state-of-the-art accuracy for latency-optimized networks on the Cityscapes semantic segmentation dataset. Our latency-optimized small SqueezeNAS network achieves 68.02% validation class mIOU with less than 35 ms inference times on the NVIDIA AGX Xavier. Our latency-optimized large SqueezeNAS network achieves 73.62% class mIOU with less than 100 ms inference times. We demonstrate that significant performance gains are possible by utilizing NAS to find networks optimized for both the specific task and inference hardware. We also present detailed analysis comparing our networks to recent state-of-the-art architectures.

Table of Contents
--------------------

- (1) Introduction and Motivation
- (2) Related Work
  - (2-1) Semantic Segmentation
  - (2-2) Efficient Network Design
  - (2-3) Neural Architecture Search (NAS)
- (3) Architecture Search Space
  - (3-1) Constrained Macro-Architecture
  - (3-2) Block Search Space
- (4) Neural Architecture Search Algorithm
  - (4-1) Gumbel-Softmax
  - (4-2) Early Stopping
  - (4-3) Resource-Aware Architecture Search
- (5) Experiments and Results
  - (5-1) Hardware-Agnostic Search
  - (5-2) Hardware-Aware Search
  - (5-3) Implementation
    - (5-3-1) Architecture Search
    - (5-3-2) Training Details
  - (5-4) Results
- (6) Network Analysis
- (7) Conclusion
- (A) XLarge Networks Visualization
- (B) Parameters of Candidate Blocks in our Search Space
- (C) Parameters of Macro Search Spaces
  - (C-1) Small Network Search Space
  - (C-2) Large Network Search Space
  - (C-3) XLarge Network Search Space

BibTeX
------

```bibtex
@misc{shaw2019squeezenas,
    title={SqueezeNAS: Fast neural architecture search for faster semantic segmentation},
    author={Albert Shaw and Daniel Hunter and Forrest Iandola and Sammy Sidhu},
    year={2019},
    eprint={1908.01748},
    archivePrefix={arXiv},
    primaryClass={cs.CV}
}
```
